android/src/main/java/com/getbouncer/RNCardVerifyModule.java

```diff
@@ -306,11 +311,44 @@ public void isSupportedAsync(Promise promise) {
     }
 
     @ReactMethod
-    public void downloadModels() {
+    public void awaitReady(@NotNull final Promise promise) {
+        if (modelsDownloaded) {
+            promise.resolve(true);
+        } else {
+            modelsDownloadPromise = promise;
+        }
+    }
+
+    @ReactMethod
+    public void downloadModels(@Nullable final Promise promise) {
         if (useLocalVerificationOnly) {
-            com.getbouncer.cardverify.ui.local.CardVerifyActivity.warmUp(this.reactContext.getApplicationContext(), apiKey, enableExpiryExtraction || enableNameExtraction, false);
+            com.getbouncer.cardverify.ui.local.CardVerifyActivity.prepareScan(this.reactContext.getApplicationContext(), apiKey, enableExpiryExtraction || enableNameExtraction, false, new Function0<Unit>() {
+                @Override
+                public Unit invoke() {
```

Comment:
It looks like the two `invoke` methods are identical, is there an opportunity for reuse?

Reply:
most likely. I was being lazy 😛